### ISSUE TITLE
It's not clear what applies to full MVC only and what can be used in ApiControllers

### ISSUE THREAD

**Issue by eparizzi** [dotnet/AspNetCore.Docs#18644]
This page list several validation methods in ASP.NET Core but fails to clarify what works in Web API, what works in MVC only and what works in both.

For example, the first section, [Model State](https://docs.microsoft.com/en-us/aspnet/core/mvc/models/validation?view=aspnetcore-3.1#model-state) says 

> Web API controllers don't have to check ModelState.IsValid if they have the [ApiController] attribute. In that case, an automatic HTTP 400 response containing error details is returned when model state is invalid. For more information, see Automatic HTTP 400 responses.

So I assume all other methods in this page apply to Web API controllers.

But then, I've been told [here](https://github.com/dotnet/AspNetCore.Docs/issues/18438) that [Top-level node validation](https://docs.microsoft.com/en-us/aspnet/core/mvc/models/validation?view=aspnetcore-3.1#top-level-node-validation) only works for full MVC controllers.

In conclusion, I think this page could explain a little bit better what validation methods work for Web API controllers too.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 0156ffc3-47ac-d5da-fbc2-9ff5018b071b
* Version Independent ID: 615db388-a244-41f7-21b5-a4085d9ed479
* Content: [Model validation in ASP.NET Core MVC](https://docs.microsoft.com/en-us/aspnet/core/mvc/models/validation?view=aspnetcore-3.1)
* Content Source: [aspnetcore/mvc/models/validation.md](https://github.com/dotnet/AspNetCore.Docs/blob/master/aspnetcore/mvc/models/validation.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-mvc**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

**Comment by Rick-Anderson** [comment]
There is no such thing a s full MVC

**Comment by eparizzi** [comment]
You're right. I've just learnt the differences between the following (not from any ASP.NET documentation page by the way).
```
services.AddMvcCore()
services.AddMvc()
services.AddControllers()
services.AddControllersWithViews()
services.AddRazorPages()
```
So in the [other issue](https://github.com/dotnet/AspNetCore.Docs/issues/18438) I asked about this your answer was "The doc is on ASP.NET Core controllers and views, not web APIs", which is not accurate.

The reason top-level node attribute validation wasn't working in my case was because I was using `.AddMvcCore()`. This feature requires the `.AddDataAnnotations()` which is already included when using `AddControllers()` and `AddControllersWithViews()`.

I had to dig around the internet and tried all the different options on a fresh project to understand this. 

I still insist that this should be made clear in the documentation.
